### ISSUE TITLE
Long running response / setTimeout not being called

### ISSUE THREAD

**Issue by stursby** [alexa-js/alexa-app#239]
What is the longest an Alexa skill can wait before responding... Here's some sample code that I've been testing out (on both Lambda & on a express / Node version too).

```javascript
const alexa = require('alexa-app')
const app = new alexa.app('my-long-running-app')

app.intent('SampleIntent', {
  'utterances': ['sample']
}, (req, res) => {
  return new Promise((resolve) => {
    setTimeout(() => {
      res.say('I waited a minute')
      resolve()
    }, 60000)
  })
})
```

The `setTimeout` is more just a placeholder for a long running function. The above code does work usually in the ~30 seconds range, but will go completely unresponsive around ~40+ seconds (And will only time out after a few minutes and say "there was a problem with the ... ").

**Comment by dblock** [comment]
I think http://stackoverflow.com/a/37264736 answers the question, which is 300 seconds, but it looks like you're seeing shorter times. I would ask some Amazon Alexa forums - would appreciate if someone could document it here as well, will leave this open.

**Comment by stursby** [comment]
@dblock Yes, I saw the AWS Lambda 300 second timeout, so that's what I set the timeout to in the AWS Dev Console... the app became unresponsive every time. I then tried using `alexa-app` with Express to see if it was maybe a Lambda issue, but the same problem occurs unfortunately.

**Comment by dblock** [comment]
You should verify whether the function actually runs.

**Comment by stursby** [comment]
@dblock what do you mean by this? What function would I be verifying?

**Comment by dblock** [comment]
I would `curl` the API endpoint and see whether the correct SSML response is returned in the long timeout case or whether there's an error or something like that.

**Comment by stursby** [comment]
@dblock I threw together a quick demo app on this URL --> https://alexa-long-response.now.sh/long-response (And you can look at the [SOURCE](https://zeit.co/stursby/alexa-long-response/lhrzgotqeq/source?f=index.js&redirect=1))

Here's an example with the utterance "wait 5 seconds"

![image](https://cloud.githubusercontent.com/assets/527849/26112975/e24202ec-3a1e-11e7-98bc-f541e90389d9.png)

And another with "wait 60 seconds"

![image](https://cloud.githubusercontent.com/assets/527849/26113010/f6de6eac-3a1e-11e7-9c15-5daeae07030d.png)

The service response in full is `The remote endpoint could not be called, or the response it returned was invalid.`

Now that you can see my `WaitSecondsIntent`, hopefully you can point me in the right direction.

Thanks!!

**Comment by stursby** [comment]
Interesting.... so if I use Postman to send a 60 second response... _totally works_ 🤔 

![image](https://cloud.githubusercontent.com/assets/527849/26113383/0410f922-3a20-11e7-8f2c-099b74c13ae2.png)

**Comment by dblock** [comment]
That's what I was getting to. The problem is not your code, it's the Alexa platform that's timing it out, I suspect it's a hard-coded limitation.

As I mentioned before, bring this up on an alexa dev forum or something like that. It will be helpful for the next person and PR an update to README here somewhere in asynchronous responses once you get a definite answer from Amazon.

**Comment by stursby** [comment]
@dblock For reference --> https://forums.developer.amazon.com/questions/71547/long-running-response-settimeout-not-being-called.html

We'll see if anyone can shed some light!

**Comment by stursby** [comment]
@dblock Just following up... got a reply in the forums

![image](https://user-images.githubusercontent.com/527849/28389471-db4f8d38-6c9b-11e7-80ab-c3802fe5deb1.png)

Which makes sense, as during testing, if the response was any longer than ~10 seconds... it would fail.

**Comment by dblock** [comment]
Can we document (and link) this in README please? Care to PR?
